pubtools/pulplib/_impl/client/client.py

```diff
@@ -302,3 +323,40 @@ def _delete_resource(self, resource_type, resource_id):
 
         LOG.debug("Queuing request to DELETE %s", url)
         return self._task_executor.submit(self._do_request, method="DELETE", url=url)
+
+    def _request_upload(self):
+        url = os.path.join(self._url, "pulp/api/v2/content/uploads/")
+
+        LOG.debug("Requesting upload id")
+        return self._request_executor.submit(self._do_request, method="POST", url=url)
+
+    def _do_upload(self, data, upload_id, offset):
+        url = os.path.join(
+            self._url, "pulp/api/v2/contents/%s/%s" % (upload_id, offset)
+        )
+
+        return self._request_executor.submit(
+            self._do_request, method="PUT", url=url, data=data
+        )
+
+    def _do_import(self, repo_id, upload_id, unit_type_id, unit_key):
+        url = os.path.join(
+            self._url, "pulp/api/v2/repositories/%s/actions/import_upload/" % repo_id
+        )
+
+        body = {
+            "unit_type_id": unit_type_id,
+            "upload_id": upload_id,
+            "unit_key": unit_key,
+        }
+
+        LOG.debug("Importing contents to repo %s", repo_id)
```

Comment:
Putting upload_id in this message would be nice.

Otherwise if you're uploading dozens of files to the same repo, all the messages will be exactly the same and you'll have no way of matching up what was handled when.